### DISCUSSION TITLE
Hold Circle

### DISCUSSION THREAD

**Discussion by home-gihub** [ppy/osu#27394] [Ideas]
A circle where you have to hold it for a short period of time like the ones in osu!stream.

**Comment by OwenCMYK** [comment]
This would be cool. Idk if they're still open to new gameplay additions though

**Reply by Gabixel** [reply]
I remember peppy saying he had some ideas for objects (example [here](https://github.com/ppy/osu/discussions/18867#discussioncomment-3029708)). Hold Circle is probably [one of them](https://github.com/ppy/osu/discussions/20325#discussioncomment-3654797), but I guess it makes the most sense for mobile (but can probably work for desktop, too, if we ignore the multitouch ability).
Also related:
- https://github.com/ppy/osu/discussions/16126#discussion-3751082

**Reply by home-gihub** [reply]
There has always been a spinner, circle and slider. so why not add a new element now?

**Comment by RaiLTC** [comment]
This could be a great replacement to extremely awkward low velocity sliders you have to make sometimes where you simply want the player to stop.. completely.

I have made a few maps before as practice on low velocity slider shapes but after making it a simple circle slider using OliBomby's Mapping Tools, I found out a simple hold circle feels better than a random shape in some cases

Besides, this can't really hurt anything as it will add more veriety and maybe even new cool gimmicks for fun, so I fully support this

**Comment by cl8n** [comment]
I have this implementation of holds from when I tried to port features from osustream

https://github.com/user-attachments/assets/abb08699-f9f3-40c1-b0b3-eec35a4ebea4

it's pretty ugly and the animations are broken, but the backing hitobject code seems to work well and could be a base for anyone who wants to try this I guess. that being said I don't think this is a good addition without multitouch mapping as in osustream, since that's where all of its unique mechanics come from.

this code was very experimental so be warned there's a lot of useless leftovers, commented code, things in the wrong place, etc... [osu branch](https://github.com/cl8n/osu/tree/hold), [osu-resources branch](https://github.com/cl8n/osu-resources/tree/hold)

**Reply by home-gihub** [reply]
Wow!

**Comment by marioman63** [comment]
buzz sliders have always been a stand in for this sort of gameplay, so a proper implementation of an actual hold note seems practical.
